### PR TITLE
[microNPU] Add support for scalar values

### PR DESCRIPTION
**Pull request by lhutton1**
PR #9515 enabled support for scalar constants, but didn't consider the case of a scalar value where the underlying constant data does not have a shape i.e. `constant.shape == []`. See the test case for a visual difference when the scalar value is 1.

### CODE REVIEW

**Comment by lhutton1** [comment]
cc @manupa-arm @mbaret @ekalda @NicolaLancellotti @jacobbohlin @dchauhan-arm

**Comment by lhutton1** [comment]
friendly ping for comment/approval

File: python/tvm/relay/backend/contrib/ethosu/tir/scheduler.py [hunk 1]
```diff
@@ -135,7 +135,17 @@ def _visit(tensor, reader, lut):
             if tensor not in planned:
                 planned.add(tensor)
                 if isinstance(tensor.op, tvm.te.PlaceholderOp) and tensor != lut:
-                    index = list(cached_func.inputs).index(tensor)
+                    # Find index of input using 'same_as' check to prevent equality
+                    # ambiguity when encountering a scalar.
+                    index = -1
+                    for i, var in enumerate(cached_func.inputs):
```
Comment (manupa-arm) [inline python/tvm/relay/backend/contrib/ethosu/tir/scheduler.py:141]: nit : could use something like : 
[input for input in cached_func.inputs if input.same_as(tensor)] and then checking the resulting list ?

**Review by manupa-arm — COMMENTED** [review]
Thanks for fixing this! just a nit to make it look simpler -- happy to take this in though! let me know.

File: python/tvm/relay/backend/contrib/ethosu/tir/scheduler.py (hunk 1 above)
Comment (lhutton1) [reply python/tvm/relay/backend/contrib/ethosu/tir/scheduler.py:141]: Agreed, I'll make the change here :)

**Review by manupa-arm — APPROVED** [review]
LGTM!

**Comment by manupa-arm** [comment]
Thanks! @lhutton1

**Pull request merged**